instrumentation/internal/internal-reflection/bootstrap/src/main/java/io/opentelemetry/javaagent/instrumentation/internal/reflection/RealInterfaces.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.internal.reflection;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+public final class RealInterfaces {
+  private static final Map<Class<?>, Class<?>[]> realInterfaces =
+      Collections.synchronizedMap(new WeakHashMap<>());
```

Comment:
I'm worried about the strong reference to the classes in the value array, can you add a comment here explaining why this is ok (if it is?)

Reply:
I initially thought that it would be ok because value doesn't reference key, but this logic doesn't work when classes are in the same class loader as value keeps the loader alive which prevents the key from being collected. I'll try to verify this tomorrow. It is probably easiest to change this to just keep the flag whether class has virtual fields or not instead of keeping the list of interfaces. Or if you prefer could also wrap all the interfaces with weakreferences.

Reply:
> It is probably easiest to change this to just keep the flag whether class has virtual fields or not instead of keeping the list of interfaces

👍